components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/internal/builder/AuthReqBuilder.java

```diff
@@ -180,11 +179,15 @@ private static Issuer buildIssuer(String issuerId) {
      *
      * @return NameIDPolicy object
      */
-    private static NameIDPolicy buildNameIDPolicy() {
-        NameIDPolicy nameIDPolicy = new NameIDPolicyBuilder().buildObject();
-        nameIDPolicy.setFormat(SSOConstants.SAML2_NAME_ID_POLICY);
-        nameIDPolicy.setAllowCreate(true);
-        return nameIDPolicy;
+    private static NameIDPolicy buildNameIDPolicy(String nameIdPolicy) {
+        NameIDPolicy nameIDPolicyObj = new NameIDPolicyBuilder().buildObject();
+        if (nameIdPolicy != null && !nameIdPolicy.isEmpty()){
+            nameIDPolicyObj.setFormat(nameIdPolicy);
+        }else {
```

Comment:
Fixed in #1581(https://github.com/wso2/carbon-apimgt/commit/ba3007eb347b47f68e1dd6e75fe8102d057200dc)